### ISSUE TITLE
[Bug]: Upgrading from v2.3.15 to master-20240625-506a9152-amd64 due to DN and QN crash and one indexnode OOM

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#34176]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:v2.3.15 to master-20240625-506a9152-amd64 
- Deployment mode(standalone or cluster):both
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2024-06-25T10:11:26.650Z] + kubectl get pods -o wide

[2024-06-25T10:11:26.651Z] + grep pulsar-cluster-upgrade-2458

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-bookie-0                              1/1     Running            0                  43m     10.104.18.34    4am-node25   <none>           <none>

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-bookie-1                              1/1     Running            0                  43m     10.104.17.153   4am-node23   <none>           <none>

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-broker-0                              1/1     Running            0                  43m     10.104.1.99     4am-node10   <none>           <none>

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-etcd-0                                1/1     Running            0                  18m     10.104.18.61    4am-node25   <none>           <none>

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-etcd-1                                1/1     Running            0                  19m     10.104.17.205   4am-node23   <none>           <none>

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-etcd-2                                1/1     Running            0                  20m     10.104.20.136   4am-node22   <none>           <none>

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-milvus-datacoord-6c79647f7-mqshc      1/1     Running            0                  20m     10.104.1.122    4am-node10   <none>           <none>

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-milvus-datanode-84df48f46-2tlkj       1/1     Running            0                  43m     10.104.1.110    4am-node10   <none>           <none>

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-milvus-datanode-84df48f46-772x6       1/1     Running            0                  43m     10.104.14.83    4am-node18   <none>           <none>

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-milvus-datanode-87b5ddc64-7kqz5       0/1     CrashLoopBackOff   8 (105s ago)       20m     10.104.13.161   4am-node16   <none>           <none>

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-milvus-indexcoord-8574bc79fd-pgp2z    1/1     Running            0                  20m     10.104.21.245   4am-node24   <none>           <none>

[2024-06-25T10:11:26.905Z] pulsar-cluster-upgrade-2458-milvus-indexnode-6f94c6f645-7l7k7     1/1     Running            1 (4m16s ago)      18m     10.104.13.166   4am-node16   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-milvus-indexnode-6f94c6f645-htmjj     1/1     Running            3 (16m ago)        20m     10.104.1.120    4am-node10   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-milvus-indexnode-6f94c6f645-mcm5j     0/1     OOMKilled          4 (93s ago)        15m     10.104.19.175   4am-node28   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-milvus-proxy-5f555f4dd6-ts55z         1/1     Running            0                  20m     10.104.1.121    4am-node10   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-milvus-querycoord-656cf75dd5-6qds6    1/1     Running            0                  20m     10.104.13.162   4am-node16   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-milvus-querynode-59d68b5b55-6wll4     0/1     CrashLoopBackOff   7 (3m56s ago)      16m     10.104.1.133    4am-node10   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-milvus-querynode-59d68b5b55-lpxp8     0/1     CrashLoopBackOff   7 (5m1s ago)       20m     10.104.21.244   4am-node24   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-milvus-querynode-5cd99499f5-gj94d     1/1     Running            0                  43m     10.104.18.27    4am-node25   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-milvus-querynode-5cd99499f5-s6sqz     1/1     Running            0                  43m     10.104.1.102    4am-node10   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-milvus-rootcoord-7cc5dd86f7-95nm7     1/1     Running            0                  20m     10.104.1.123    4am-node10   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-minio-0                               1/1     Running            0                  43m     10.104.17.152   4am-node23   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-minio-1                               1/1     Running            0                  43m     10.104.18.36    4am-node25   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-minio-2                               1/1     Running            0                  43m     10.104.25.227   4am-node30   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-minio-3                               1/1     Running            0                  43m     10.104.20.133   4am-node22   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-proxy-0                               1/1     Running            0                  43m     10.104.1.101    4am-node10   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-recovery-0                            1/1     Running            0                  43m     10.104.14.79    4am-node18   <none>           <none>

[2024-06-25T10:11:26.906Z] pulsar-cluster-upgrade-2458-zookeeper-0                           1/1     Running            0                  43m     10.104.18.33    4am-node25   <none>           <none>
```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/deploy_test_cron/detail/deploy_test_cron/2458/pipeline
log:
[artifacts-pulsar-cluster-upgrade-2458-server-logs (1).tar.gz](https://github.com/user-attachments/files/15981806/artifacts-pulsar-cluster-upgrade-2458-server-logs.1.tar.gz)

**Comment by smellthemoon** [comment]
/assign
<img width="987" alt="image" src="https://github.com/milvus-io/milvus/assets/64083300/33d0ae8b-7a7c-4f7b-a79b-a3df81a872a5">

**Comment by zhuwenxing** [comment]
verified and fix in `master-20240709-eeb03a0d-amd64`
